Framework/PythonInterface/plugins/algorithms/SaveHKLCW.py

```diff
@@ -70,13 +70,8 @@ def PyExec(self):
 
             if directionCosines:
                 U = peak_ws.sample().getOrientedLattice().getU()
-                sample_pos = peak_ws.getInstrument().getSample().getPos()
-                source_pos = peak_ws.getInstrument().getSource().getPos()
-                ki_n = sample_pos - source_pos  # direction of incident wavevector
-                ki_n = ki_n * (1. / ki_n.norm())
-
-            for p in peak_ws:
-                if directionCosines:
+                for p in peak_ws:
+                    ki_n = p.getReferenceFrame().vecPointingAlongBeam()
```

Comment:
This works only if the sample's position has not been calibrated, thus stays at the origin. Is this fair to assume?